src/hotspot/share/opto/memnode.cpp

```diff
@@ -1625,7 +1625,13 @@ Node *LoadNode::split_through_phi(PhaseGVN *phase) {
       the_clone = x;            // Remember for possible deletion.
       // Alter data node to use pre-phi inputs
       if (this->in(0) == region) {
-        x->set_req(0, in);
+        if (mem->is_Phi() && (mem->in(0) == region) && mem->in(i)->in(0) != NULL) {
```

Comment:
I have concern about this because you can have load's address dependency on a control node below memory's control.
I saw cases when Load's address and memory control were Region through which it splits. And all controls were set  to Region's inputs.
Also consider that we did not split memory node yet. We may end up with load's clone placed above its memory.